File: beginner/python/reverse-array-in-place.py
"""
  Create a function that reverses the array passed in.
  It is important that these two rules are kept:
    1. Do not create a new array
    2. Do not use the native array reverse method
"""

import math

def reverse_array_in_place(arr):
  half_len = int(math.ceil(len(arr)/2))
  for index in range(half_len):
    end_index = len(arr) - 1 - index
    # dont need temp vars in python, can do multiple assignment
    arr[index], arr[end_index] = arr[end_index], arr[index]

  return arr

print(reverse_array_in_place([1, 3, 5, 7, 9]))
print(reverse_array_in_place([1, 3, 5, 7, 9, 11]))

File: beginner/javascript/reverse-array-in-place.js
/**
  Create a function that reverses the array passed in.
  It is important that these two rules are kept:
    1. Do not create a new array
    2. Do not use the native array reverse method
 */

function reverseArrayInPlace(arr) {
  for (let i = 0; i < arr.length / 2; i++) {
    const temp = arr[i];
    const lastIndex = arr.length - 1 - i;
    arr[i] = arr[lastIndex];
    arr[lastIndex] = temp;
  }
  return arr;
}

console.log(reverseArrayInPlace([1, 3, 5, 7, 9]));
console.log(reverseArrayInPlace([1, 3, 5, 7, 9, 11]));
